lib/spark.js

```diff
@@ -0,0 +1,98 @@
+import { assertEquals } from 'zinnia:assert'
+
+const sleep = dt => new Promise(resolve => setTimeout(resolve, dt))
+
+export default class Spark {
+  async getRetrieval () {
+    console.log('Geting retrieval...')
+    const res = await fetch('https://spark.fly.dev/retrievals', {
+      method: 'POST'
+    })
+    const retrieval = await res.json()
+    console.log({ retrieval })
+    return retrieval
+  }
+  
+  async fetchCAR (url) {
+    const stats = {
+      start: new Date(),
+      firstByte: null,
+      end: null,
+      byteLength: 0,
+      status: null
+    }
+    console.log('Fetching CAR...')
+    const controller = new AbortController()
+    const { signal } = controller
+    const res = await fetch(url, { signal })
+    stats.status = res.status
+  
+    if (res.ok) {
+      // Abort if no data is received for 10 seconds
+      let timeout
+      const startTimeout = () => {
+        if (timeout) {
+          clearTimeout(timeout)
+        }
+        timeout = setTimeout(() => controller.abort(), 10_000)
+      }
+      startTimeout()
+  
+      try {
+        for await (const value of res.body) {
+          if (stats.firstByte === null) {
+            stats.firstByte = new Date()
+          }
+          stats.byteLength += value.byteLength
+          startTimeout()
+        }
+      } finally {
+        clearTimeout(timeout)
+      }
+    }
+    stats.end = new Date()
+    console.log(stats)
+  }
+  
+  async submitRetrieval ({ id, success }) {
+    console.log('Submitting retrieval...')
+    const res = await fetch(`https://spark.fly.dev/retrievals/${id}`, {
+      method: 'PATCH',
+      body: JSON.stringify({ success }),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    })
+    assertEquals(res.status, 200, await res.text().catch(() => ''))
+    console.log('Retrieval submitted')
+  }
+
+  async nextRetrieval () {
+    const retrieval = await this.getRetrieval()
+
+    let success = false
+    const url = `https://strn.pl/ipfs/${retrieval.cid}`
+    try {
+      await this.fetchCAR(url)
+      success = true
+    } catch (err) {
+      console.error(`Failed to fetch ${url}`)
+      console.error(err)
+    }
+
+    await this.submitRetrieval({ id: retrieval.id, success })
+    Zinnia.jobCompleted()
```

Comment:
@bajtos what do you think about removing this global in favor of a module import? Or deprecating it rather. I'd prefer if any variable that isn't web standard be explicitly imported

```js
import { Zinnia } from `zinnia`
```

Reply:
@juliangruber Could you please open an issue in Zinnia for this enhancement?

Reply:
https://github.com/filecoin-station/zinnia/issues/236